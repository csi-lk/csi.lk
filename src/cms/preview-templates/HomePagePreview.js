import React from 'react'
import PropTypes from 'prop-types'

import { HomePageTemplate } from '../../templates/home/home'

const HomePagePreview = ({ widgetFor }) => (
  <HomePageTemplate
    content={widgetFor('body')}
  />
)


HomePagePreview.propTypes = {
  widgetFor: PropTypes.func.isRequired,
}

export default HomePagePreview
