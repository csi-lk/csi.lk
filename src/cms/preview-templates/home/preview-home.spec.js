import React from 'react'

import HomePagePreview from './preview-home'

describe('Preview Template: Home', () => {
  it('renders', () => {
    const wrapper = shallow(<HomePagePreview widgetFor={() => 'test'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
