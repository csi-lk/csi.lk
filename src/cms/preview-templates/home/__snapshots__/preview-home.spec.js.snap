// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Template: Home renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePagePreview
    widgetFor={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "data": Object {
        "articles": Object {
          "edges": Array [
            Object {
              "node": Object {
                "frontmatter": Object {
                  "path": "/test",
                  "tags": "one, two",
                  "title": "Test",
                },
                "timeToRead": 1,
              },
            },
          ],
        },
        "content": Object {
          "frontmatter": Object {
            "description": "test",
            "title": "test",
          },
          "html": "test",
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "articles": Object {
            "edges": Array [
              Object {
                "node": Object {
                  "frontmatter": Object {
                    "path": "/test",
                    "tags": "one, two",
                    "title": "Test",
                  },
                  "timeToRead": 1,
                },
              },
            ],
          },
          "content": Object {
            "frontmatter": Object {
              "description": "test",
              "title": "test",
            },
            "html": "test",
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
